streams/src/main/java/org/apache/kafka/streams/state/internals/TransactionalTimestampedSegment.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import java.io.File;
+import java.util.Map;
+import org.apache.kafka.common.utils.Bytes;
+
+public class TransactionalTimestampedSegment extends AbstractTransactionalSegment {
+    TransactionalTimestampedSegment(final KeyValueSegment tmpStore,
+                                    final TimestampedSegment mainStore) {
+        super(tmpStore, mainStore);
+    }
+
+    @Override
+    public void deleteRange(final Bytes keyFrom, final Bytes keyTo) {
+        throw new UnsupportedOperationException();
```

Comment:
This one will be tricky since you'll have to do a normal deleteRange on the tmpStore, then scan over the same range in the committed store and delete those keys in the tmp store as well.